tests/test_treon.py

```diff
@@ -0,0 +1,33 @@
+from treon import treon
+
+
+def test_filter_results():
+    args = {
+        "--exclude": ['resources/basic.ipynb',
+                      'failed']
+    }
+    results = ['resources/basic.ipynb',
+               'resources/doctest_failed.ipynb',
+               'resources/runtime_error.ipynb',
+               'resources/unittest_failed.ipynb',
+               'other/resources.ipynb']
+
+    filtered = treon.filter_results(results=results, args=args)
+    expected = ['resources/doctest_failed.ipynb',
+                'resources/runtime_error.ipynb',
+                'resources/unittest_failed.ipynb',
+                'other/resources.ipynb']
+    assert filtered == expected
```

Comment:
It would be clean to write 3 separate tests, one for each assert you have - 

def exclude_option_works_with_file_paths
def exclude_option_works_with_directory_paths
def exclude_option_works_with_file_and_directory_paths_combined

that way you can move the results = [ ... ] into the setup method.